src/webui/www/private/scripts/torrent-content.js

```diff
@@ -0,0 +1,572 @@
+        });
+
+        torrentFilesTable.populateTable(rootNode);
+        torrentFilesTable.updateTable();
+
+        if (selectedFiles.length > 0)
+            torrentFilesTable.reselectRows(selectedFiles);
+    };
+
+    const filesPriorityMenuClicked = (priority) => {
+        const selectedRows = torrentFilesTable.selectedRowsIds();
+        if (selectedRows.length === 0)
+            return;
+
+        const rowIds = [];
+        const fileIds = [];
+        selectedRows.forEach((rowId) => {
+            rowIds.push(rowId);
+            fileIds.push(Number(torrentFilesTable.getRowFileId(rowId)));
+        });
+
+        const uniqueRowIds = new Set();
+        const uniqueFileIds = new Set();
+        for (let i = 0; i < rowIds.length; ++i) {
+            const rows = getAllChildren(rowIds[i], fileIds[i]);
+            rows.rowIds.forEach((rowId) => {
+                uniqueRowIds.add(rowId);
+            });
+            rows.fileIds.forEach((fileId) => {
+                uniqueFileIds.add(fileId);
+            });
+        }
+
+        setFilePriority([...uniqueRowIds.keys()], [...uniqueFileIds.keys()], priority);
+        for (const id of rowIds)
+            updateParentFolder(id);
+    };
+
+    const updateParentFolder = (id) => {
+        const updateComplete = () => {
+            // we've finished recursing
+            updateGlobalCheckbox();
+            torrentFilesTable.updateTable(true);
+        };
+
+        const node = torrentFilesTable.getNode(id);
+        const parent = node.parent;
+        if (parent === torrentFilesTable.getRoot()) {
+            updateComplete();
+            return;
+        }
+
+        const siblings = parent.children;
+
+        let checkedCount = 0;
+        let uncheckedCount = 0;
+        let indeterminateCount = 0;
+        let desiredComboboxPriority = null;
+        for (const sibling of siblings) {
+            switch (getCheckboxState(sibling.rowId)) {
+                case TriState.Checked:
+                    checkedCount++;
+                    break;
+                case TriState.Unchecked:
+                    uncheckedCount++;
+                    break;
+                case TriState.Partial:
+                    indeterminateCount++;
+                    break;
+            }
+
+            if (desiredComboboxPriority === null)
+                desiredComboboxPriority = getComboboxPriority(sibling.rowId);
+            else if (desiredComboboxPriority !== getComboboxPriority(sibling.rowId))
+                desiredComboboxPriority = FilePriority.Mixed;
+        }
+
+        const currentCheckboxState = getCheckboxState(parent.rowId);
+        let desiredCheckboxState;
+        if ((indeterminateCount > 0) || ((checkedCount > 0) && (uncheckedCount > 0)))
+            desiredCheckboxState = TriState.Partial;
+        else if (checkedCount > 0)
+            desiredCheckboxState = TriState.Checked;
+        else
+            desiredCheckboxState = TriState.Unchecked;
```

Comment:
```suggestion
        let desiredCheckboxState = TriState.Unchecked;
        if ((indeterminateCount > 0) || ((checkedCount > 0) && (uncheckedCount > 0)))
            desiredCheckboxState = TriState.Partial;
        else if (checkedCount > 0)
            desiredCheckboxState = TriState.Checked;
```

Reply:
Also moved code so will leave for a separate PR

Reply:
>Also moved code so will leave for a separate PR

Could you point out where was the code in master?

Reply:
Was mistaken here as well, this is new code. Will update.